src/Analyzer.Cli/CommandLineParser.cs

```diff
@@ -190,40 +172,27 @@ private void SetupCommonOptionsForCommands(List<Command> commands)
             }
 
             // Verify the file is a valid template
-            if (!IsValidTemplate(templateFilePath))
+            if (!TemplateDiscovery.IsValidTemplate(templateFilePath))
```

Comment:
`TemplateDiscovery` is most useful for searching a directory for templates and finding matching parameters.  Since `TemplateProcessor` is only designed to process a template (which is already discovered), I don't think that's a good place for it.

`Utilities` might be a good place for it.  Need to think through that more.